### PR TITLE
feat: persistindo dados do usuário ao logar

### PR DESCRIPTION
**Pull request by vanildojr33**
Adição do AsyncStorage para ser possível acessar os dados do usuário em outras telas.

### CODE REVIEW

File: NaReguaApp/src/routes/index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	const { logado, isProvider } = useContext(AuthContext);
+
+	return (
+		<>
+			{logado ?
+				<>
+					{isProvider ? <AppClienteRoute /> : <AppPrestadorRoute />}
+				</>
+				: <AuthRoute />
+			}
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:19]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······)`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      )}
```

File: NaReguaApp/src/routes/app-cliente.routes.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
+import InicioCliente from '../paginas/inicio_cliente';
+
+const ClienteStack = createNativeStackNavigator();
+
+function AppClienteRoute() {
+  return (
+    <ClienteStack.Navigator initialRouteName="InicioCliente">
+        <ClienteStack.Screen name="InicioCliente" component={InicioCliente}/>
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/app-cliente.routes.js:10]: Codacy has a fix for the issue: [Replace `··<ClienteStack.Screen·name="InicioCliente"·component={InicioCliente}` with `<ClienteStack.Screen·name="InicioCliente"·component={InicioCliente}·`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      <ClienteStack.Screen name="InicioCliente" component={InicioCliente} />
```

File: NaReguaApp/src/contexts/auth.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+	useEffect(() => {
+		async function carregarStorage() {
+			const storageUser = await AsyncStorage.getItem('user');
+			const storageToken = await AsyncStorage.getItem('token');
+
+			if (storageUser && storageToken) {
+				setUser(JSON.parse(storageUser));
+				setIsProvider(storageUser.isProvider);
+				api.defaults.headers.common['Authorization'] = `Bearer ${storageToken}`;
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:20]: Codacy has a fix for the issue: [Replace `↹↹↹↹` with `········`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
        api.defaults.headers.common['Authorization'] = `Bearer ${storageToken}`;
```

File: NaReguaApp/src/contexts/auth.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+		} catch (error) {
+				console.log(error);
+				if(error.response) {
+					setMensagemErro(error.response.data.message);
+				}
+		}
+	}
+
+	return (
+		<AuthContext.Provider value={{ logado: !!user, isProvider, user, login, mensagemErro }}>
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:49]: Codacy has a fix for the issue: [Replace `↹↹<AuthContext.Provider·value={{·logado:·!!user,·isProvider,·user,·login,·mensagemErro·` with `····<AuthContext.Provider⏎······value={{logado:·!!user,·isProvider,·user,·login,·mensagemErro`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
    <AuthContext.Provider
```

File: NaReguaApp/src/contexts/auth.js (hunk 3 above)
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:20]: Codacy has a fix for the issue: [["Authorization"] is better written in dot notation.](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
				api.defaults.headers.common.Authorization = `Bearer ${storageToken}`;
```

File: NaReguaApp/src/contexts/auth.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	}, []);
+
+	async function login(data) {
+		try {
+			let res = await api.post('/login', data);
+
+			let { token } = res.data;
+			let responseUser = res.data.user;
+
+			api.defaults.headers.common['Authorization'] = `Bearer ${token}`;
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:34]: Codacy has a fix for the issue: [["Authorization"] is better written in dot notation.](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
			api.defaults.headers.common.Authorization = `Bearer ${token}`;
```

File: NaReguaApp/src/contexts/auth.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import api from '../services/api';
+
+const AuthContext = createContext({});
+
+export const AuthProvider = ({ children }) => {
+	const [user, setUser] = useState(null);
+	const [isProvider, setIsProvider] = useState(false);
+	const [mensagemErro, setMensagemErro] = useState('');
+
+	useEffect(() => {
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:12]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
  useEffect(() => {
```

File: NaReguaApp/src/routes/index.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React, { useContext } from 'react';
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:1]: Codacy has a fix for the issue: [Replace `·useContext·` with `useContext`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
import React, {useContext} from 'react';
```

File: NaReguaApp/src/contexts/auth.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+		async function carregarStorage() {
+			const storageUser = await AsyncStorage.getItem('user');
+			const storageToken = await AsyncStorage.getItem('token');
+
+			if (storageUser && storageToken) {
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:17]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      if (storageUser && storageToken) {
```

File: NaReguaApp/src/contexts/auth.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import React, { createContext, useState, useEffect } from 'react';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import api from '../services/api';
+
+const AuthContext = createContext({});
+
+export const AuthProvider = ({ children }) => {
+	const [user, setUser] = useState(null);
+	const [isProvider, setIsProvider] = useState(false);
+	const [mensagemErro, setMensagemErro] = useState('');
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:10]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
  const [mensagemErro, setMensagemErro] = useState('');
```

File: NaReguaApp/src/contexts/auth.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+				api.defaults.headers.common['Authorization'] = `Bearer ${storageToken}`;
+			}
+		}
+
+		carregarStorage();
+	}, []);
+
+	async function login(data) {
+		try {
+			let res = await api.post('/login', data);
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:29]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      let res = await api.post('/login', data);
```

File: NaReguaApp/src/routes/index.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+	return (
+		<>
+			{logado ?
+				<>
+					{isProvider ? <AppClienteRoute /> : <AppPrestadorRoute />}
+				</>
+				: <AuthRoute />
+			}
+		</>
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:20]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
    </>
```

File: NaReguaApp/src/contexts/auth.js [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+				if(error.response) {
+					setMensagemErro(error.response.data.message);
+				}
+		}
+	}
+
+	return (
+		<AuthContext.Provider value={{ logado: !!user, isProvider, user, login, mensagemErro }}>
+			{children}
+		</AuthContext.Provider>
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:51]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
    </AuthContext.Provider>
```

File: NaReguaApp/src/routes/index.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+import AuthContext from '../contexts/auth';
+
+import AppClienteRoute from './app-cliente.routes';
+import AppPrestadorRoute from './app-prestator.routes';
+import AuthRoute from './auth.routes';
+
+function Routes() {
+	const { logado, isProvider } = useContext(AuthContext);
+
+	return (
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:12]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
  return (
```

File: NaReguaApp/src/contexts/auth.js [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+			api.defaults.headers.common['Authorization'] = `Bearer ${token}`;
+			await AsyncStorage.setItem('token', token);
+			await AsyncStorage.setItem('user', JSON.stringify(responseUser));
+			
+			setUser(responseUser);
+			setIsProvider(responseUser.isProvider);
+		} catch (error) {
+				console.log(error);
+				if(error.response) {
+					setMensagemErro(error.response.data.message);
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:43]: Codacy has a fix for the issue: [Replace `↹↹↹↹↹` with `········`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
        setMensagemErro(error.response.data.message);
```

File: NaReguaApp/src/contexts/auth.js [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+
+			api.defaults.headers.common['Authorization'] = `Bearer ${token}`;
+			await AsyncStorage.setItem('token', token);
+			await AsyncStorage.setItem('user', JSON.stringify(responseUser));
+			
+			setUser(responseUser);
+			setIsProvider(responseUser.isProvider);
+		} catch (error) {
+				console.log(error);
+				if(error.response) {
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:42]: Codacy has a fix for the issue: [Expected space(s) after "if".](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
				if (error.response) {
```

File: NaReguaApp/src/routes/app-prestator.routes.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+import InicioPrestador from '../paginas/inicio_prestador';
+import CadastroServico from '../paginas/cadastro_servico';
+
+const PrestadorStack = createNativeStackNavigator();
+
+function AppPrestadorRoute() {
+  return (
+    <PrestadorStack.Navigator initialRouteName="InicioPrestador">
+      <PrestadorStack.Screen name="InicioPrestador" component={InicioPrestador} options={{ headerShown: false }}/>
+      <PrestadorStack.Screen name="CadastroServico" component={CadastroServico} options={{ title: 'Novo Serviço'}}/>
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/app-prestator.routes.js:12]: Codacy has a fix for the issue: [Replace `·name="CadastroServico"·component={CadastroServico}·options={{·title:·'Novo·Serviço'}}` with `⏎········name="CadastroServico"⏎········component={CadastroServico}⏎········options={{title:·'Novo·Serviço'}}⏎······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      <PrestadorStack.Screen
```

File: NaReguaApp/src/contexts/auth.js [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+
+			if (storageUser && storageToken) {
+				setUser(JSON.parse(storageUser));
+				setIsProvider(storageUser.isProvider);
+				api.defaults.headers.common['Authorization'] = `Bearer ${storageToken}`;
+			}
+		}
+
+		carregarStorage();
+	}, []);
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:25]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
  }, []);
```

File: NaReguaApp/src/contexts/auth.js [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+	const [isProvider, setIsProvider] = useState(false);
+	const [mensagemErro, setMensagemErro] = useState('');
+
+	useEffect(() => {
+		async function carregarStorage() {
+			const storageUser = await AsyncStorage.getItem('user');
+			const storageToken = await AsyncStorage.getItem('token');
+
+			if (storageUser && storageToken) {
+				setUser(JSON.parse(storageUser));
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:18]: Codacy has a fix for the issue: [Replace `↹↹↹↹` with `········`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
        setUser(JSON.parse(storageUser));
```

File: NaReguaApp/App.js [hunk 19]
```diff
@@ -1,18 +1,12 @@
...
- * https://github.com/facebook/react-native
- *
- * @format
- * @flow strict-local
- */
-
 import React from 'react';
-import Route from './src/routes/route';
+import Routes from './src/routes/index';
+import { NavigationContainer } from '@react-navigation/native'
```
Comment (alessandrojr1998) [inline NaReguaApp/App.js:3]: Codacy has a fix for the issue: [Replace `·NavigationContainer·}·from·'@react-navigation/native'` with `NavigationContainer}·from·'@react-navigation/native';`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
import {NavigationContainer} from '@react-navigation/native';
```

File: NaReguaApp/src/contexts/auth.js [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+
+	async function login(data) {
+		try {
+			let res = await api.post('/login', data);
+
+			let { token } = res.data;
+			let responseUser = res.data.user;
+
+			api.defaults.headers.common['Authorization'] = `Bearer ${token}`;
+			await AsyncStorage.setItem('token', token);
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:35]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      await AsyncStorage.setItem('token', token);
```

File: NaReguaApp/src/contexts/auth.js [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+			await AsyncStorage.setItem('user', JSON.stringify(responseUser));
+			
+			setUser(responseUser);
+			setIsProvider(responseUser.isProvider);
+		} catch (error) {
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:40]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
    } catch (error) {
```

File: NaReguaApp/src/contexts/auth.js [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+
+			if (storageUser && storageToken) {
+				setUser(JSON.parse(storageUser));
+				setIsProvider(storageUser.isProvider);
+				api.defaults.headers.common['Authorization'] = `Bearer ${storageToken}`;
+			}
+			setLoading(false);
+		}
+
+		carregarStorage();
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:26]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
    carregarStorage();
```

File: NaReguaApp/src/routes/index.js [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+		<>
+			{logado ?
+				<>
+					{isProvider ? <AppClienteRoute /> : <AppPrestadorRoute />}
+				</>
+				: <AuthRoute />
+			}
+		</>
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:29]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
    </>
```

File: NaReguaApp/src/routes/auth.routes.js [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+import Cadastro from '../paginas/cadastro';
+import Login from '../paginas/login';
+
+const AuthStack = createNativeStackNavigator();
+
+function AuthRoute() {
+  return (
+    <AuthStack.Navigator>
+      <AuthStack.Screen name="Login" component={Login} options={{ headerShown: false }}/>
+      <AuthStack.Screen name="Cadastro" component={Cadastro} options={{ headerShown: false }}/>
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/auth.routes.js:12]: Codacy has a fix for the issue: [Replace `·name="Cadastro"·component={Cadastro}·options={{·headerShown:·false·}}` with `⏎········name="Cadastro"⏎········component={Cadastro}⏎········options={{headerShown:·false}}⏎······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      <AuthStack.Screen
```

File: NaReguaApp/src/routes/index.js [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+import { useAuth } from '../contexts/auth';
+
+import AppClienteRoute from './app-cliente.routes';
+import AppPrestadorRoute from './app-prestator.routes';
+import AuthRoute from './auth.routes';
+
+function Routes() {
+	const { logado, isProvider, loading } = useAuth();
+
+	if (loading) {
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:13]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
  if (loading) {
```

File: NaReguaApp/src/routes/index.js [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+		return (
+			<View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+				<ActivityIndicator size="large" color="#666" />
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:16]: Codacy has a fix for the issue: [Replace `↹↹↹↹` with `········`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
        <ActivityIndicator size="large" color="#666" />
```

File: NaReguaApp/src/routes/auth.routes.js [hunk 27]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/auth.routes.js:2]: Codacy has a fix for the issue: [Replace `·createNativeStackNavigator·` with `createNativeStackNavigator`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
import {createNativeStackNavigator} from '@react-navigation/native-stack';
```

File: NaReguaApp/src/contexts/auth.js [hunk 28]
```diff
@@ -0,0 +1,64 @@
...
+
+	useEffect(() => {
+		async function carregarStorage() {
+			const storageUser = await AsyncStorage.getItem('user');
+			const storageToken = await AsyncStorage.getItem('token');
+
+			if (storageUser && storageToken) {
+				setUser(JSON.parse(storageUser));
+				setIsProvider(storageUser.isProvider);
+				api.defaults.headers.common['Authorization'] = `Bearer ${storageToken}`;
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:21]: Codacy has a fix for the issue: [["Authorization"] is better written in dot notation.](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
				api.defaults.headers.common.Authorization = `Bearer ${storageToken}`;
```

File: NaReguaApp/src/contexts/auth.js [hunk 29]
```diff
@@ -0,0 +1,64 @@
...
+		<AuthContext.Provider value={{ logado: !!user, isProvider, user, loading, login, mensagemErro }}>
+			{children}
+		</AuthContext.Provider>
+	)
+}
+
+export function useAuth() {
+	const context = useContext(AuthContext);
+	
+	return context;
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:63]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
  return context;
```

File: NaReguaApp/src/contexts/auth.js [hunk 30]
```diff
@@ -0,0 +1,64 @@
...
+
+export const AuthProvider = ({ children }) => {
+	const [user, setUser] = useState(null);
+	const [isProvider, setIsProvider] = useState(false);
+	const [mensagemErro, setMensagemErro] = useState('');
+	const [loading, setLoading] = useState(true);
+
+	useEffect(() => {
+		async function carregarStorage() {
+			const storageUser = await AsyncStorage.getItem('user');
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:15]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      const storageUser = await AsyncStorage.getItem('user');
```

File: NaReguaApp/src/routes/index.js [hunk 31]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { View, ActivityIndicator } from 'react-native';
+
+import { useAuth } from '../contexts/auth';
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:4]: Codacy has a fix for the issue: [Replace `·useAuth·` with `useAuth`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
import {useAuth} from '../contexts/auth';
```

File: NaReguaApp/src/routes/app-cliente.routes.js [hunk 32]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/app-cliente.routes.js:2]: Codacy has a fix for the issue: [Replace `·createNativeStackNavigator·` with `createNativeStackNavigator`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
import {createNativeStackNavigator} from '@react-navigation/native-stack';
```

File: NaReguaApp/src/contexts/auth.js [hunk 33]
```diff
@@ -0,0 +1,64 @@
...
+					setMensagemErro(error.response.data.message);
+				}
+		}
+	}
+
+	return (
+		<AuthContext.Provider value={{ logado: !!user, isProvider, user, loading, login, mensagemErro }}>
+			{children}
+		</AuthContext.Provider>
+	)
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:57]: Codacy has a fix for the issue: [Replace `↹)` with `··);`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
  );
```

File: NaReguaApp/src/contexts/auth.js [hunk 34]
```diff
@@ -0,0 +1,64 @@
...
+			}
+			setLoading(false);
+		}
+
+		carregarStorage();
+	}, []);
+
+	async function login(data) {
+		try {
+			setLoading(true);
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:31]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      setLoading(true);
```

File: NaReguaApp/src/contexts/auth.js [hunk 35]
```diff
@@ -0,0 +1,64 @@
...
+			let res = await api.post('/login', data);
+
+			let { token } = res.data;
+			let responseUser = res.data.user;
+
+			api.defaults.headers.common['Authorization'] = `Bearer ${token}`;
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:37]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      api.defaults.headers.common['Authorization'] = `Bearer ${token}`;
```

File: NaReguaApp/src/routes/app-prestator.routes.js (hunk 16 above)
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/app-prestator.routes.js:12]: Codacy has a fix for the issue: [Replace `·name="CadastroServico"·component={CadastroServico}·options={{·title:·'Novo·Serviço'}}` with `⏎········name="CadastroServico"⏎········component={CadastroServico}⏎········options={{title:·'Novo·Serviço'}}⏎······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      <PrestadorStack.Screen
```

File: NaReguaApp/src/contexts/auth.js [hunk 36]
```diff
@@ -0,0 +1,64 @@
...
+				console.log(error);
+				if(error.response) {
+					setMensagemErro(error.response.data.message);
+				}
+		}
+	}
+
+	return (
+		<AuthContext.Provider value={{ logado: !!user, isProvider, user, loading, login, mensagemErro }}>
+			{children}
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:55]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      {children}
```

File: NaReguaApp/src/contexts/auth.js [hunk 37]
```diff
@@ -0,0 +1,64 @@
...
+
+		carregarStorage();
+	}, []);
+
+	async function login(data) {
+		try {
+			setLoading(true);
+			let res = await api.post('/login', data);
+
+			let { token } = res.data;
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:34]: Codacy has a fix for the issue: [Replace `↹↹↹let·{·token·` with `······let·{token`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      let {token} = res.data;
```

File: NaReguaApp/src/routes/auth.routes.js [hunk 38]
```diff
@@ -0,0 +1,17 @@
...
+import Cadastro from '../paginas/cadastro';
+import Login from '../paginas/login';
+
+const AuthStack = createNativeStackNavigator();
+
+function AuthRoute() {
+  return (
+    <AuthStack.Navigator>
+      <AuthStack.Screen name="Login" component={Login} options={{ headerShown: false }}/>
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/auth.routes.js:11]: Codacy has a fix for the issue: [Replace `·name="Login"·component={Login}·options={{·headerShown:·false·}}` with `⏎········name="Login"⏎········component={Login}⏎········options={{headerShown:·false}}⏎······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      <AuthStack.Screen
```

File: NaReguaApp/src/routes/index.js [hunk 39]
```diff
@@ -0,0 +1,33 @@
...
+		)
+	}
+
+	return (
+		<>
+			{logado ?
+				<>
+					{isProvider ? <AppClienteRoute /> : <AppPrestadorRoute />}
+				</>
+				: <AuthRoute />
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:27]: Codacy has a fix for the issue: [Replace `↹↹↹↹:·` with `······)·:·(⏎········`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      ) : (
```

File: NaReguaApp/src/routes/index.js [hunk 40]
```diff
@@ -0,0 +1,33 @@
...
+
+import AppClienteRoute from './app-cliente.routes';
+import AppPrestadorRoute from './app-prestator.routes';
+import AuthRoute from './auth.routes';
+
+function Routes() {
+	const { logado, isProvider, loading } = useAuth();
+
+	if (loading) {
+		return (
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:14]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
  return (
```

File: NaReguaApp/src/routes/index.js [hunk 41]
```diff
@@ -0,0 +1,33 @@
...
+			<View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+				<ActivityIndicator size="large" color="#666" />
+			</View>
+		)
+	}
```
Comment (alessandrojr1998) [inline NaReguaApp/src/routes/index.js:19]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······)`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      )}
```

File: NaReguaApp/src/contexts/auth.js [hunk 42]
```diff
@@ -0,0 +1,64 @@
...
+export const AuthProvider = ({ children }) => {
+	const [user, setUser] = useState(null);
+	const [isProvider, setIsProvider] = useState(false);
+	const [mensagemErro, setMensagemErro] = useState('');
+	const [loading, setLoading] = useState(true);
+
+	useEffect(() => {
+		async function carregarStorage() {
+			const storageUser = await AsyncStorage.getItem('user');
+			const storageToken = await AsyncStorage.getItem('token');
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:16]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
      const storageToken = await AsyncStorage.getItem('token');
```

File: NaReguaApp/src/contexts/auth.js [hunk 43]
```diff
@@ -0,0 +1,64 @@
+import React, { createContext, useState, useEffect, useContext } from 'react';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import api from '../services/api';
+
+const AuthContext = createContext({});
+
+export const AuthProvider = ({ children }) => {
+	const [user, setUser] = useState(null);
+	const [isProvider, setIsProvider] = useState(false);
```
Comment (alessandrojr1998) [inline NaReguaApp/src/contexts/auth.js:9]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/UFAPE-Projetao20202/naregua_front/pullRequest?prid=8386487)
```suggestion
  const [isProvider, setIsProvider] = useState(false);
```

**Review by eduardaluna — CHANGES_REQUESTED** [review]
OK - uso de de autenticação no login
OK - carregamento enquanto aplicação aguarda resultado da autenticação
OK - uso de stackNavigator para a navegação pelo app

**Review by eduardaluna — APPROVED** [review]

**Pull request merged**